tests/utils.go

```diff
@@ -0,0 +1,172 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"kubevirt.io/kubevirt/pkg/kubecli"
+	"kubevirt.io/kubevirt/pkg/log"
+
+)
+
+var KubeVirtVersionTag = "latest"
+var KubeVirtRepoPrefix = "kubevirt"
+var KubeVirtKubectlPath = ""
+var KubeVirtOcPath = ""
+var KubeVirtInstallNamespace = "kube-system"
+
+func init() {
+	flag.StringVar(&KubeVirtVersionTag, "tag", "latest", "Set the image tag or digest to use")
+	flag.StringVar(&KubeVirtRepoPrefix, "prefix", "kubevirt", "Set the repository prefix for all images")
+	flag.StringVar(&KubeVirtKubectlPath, "kubectl-path", "", "Set path to kubectl binary")
+	flag.StringVar(&KubeVirtOcPath, "oc-path", "", "Set path to oc binary")
+	flag.StringVar(&KubeVirtInstallNamespace, "installed-namespace", "kube-system", "Set the namespace KubeVirt is installed in")
+}
+
+const (
+	// tests.NamespaceTestDefault is the default namespace, to test non-infrastructure related KubeVirt objects.
+	NamespaceTestDefault = "cdi-test-default"
+	// NamespaceTestAlternative is used to test controller-namespace independency.
+	NamespaceTestAlternative = "cdi-test-alternative"
+)
+
+const (
+	SecretLabel = "kubevirt.io/secret"
+)
+
+var testNamespaces = []string{NamespaceTestDefault, NamespaceTestAlternative}
```

Comment:
prefer to see each test's ns be random, like done in kubernetes and kubervirt-ansible.

Reply:
Are there only 2 test namespaces?

Reply:
Yeah, at this point there are only 2. One main one, and a secondary one for the clone between namespaces tests. The name space gets cleaned out between tests.

Reply:
I see.  What do you think about dynamically generating namespace names similar ho how k8s does?  The goal being that each test spec (`It()`) would get it's own namespace.  The benefit of this would be to provide isolation for tests when running in parallel.

Reply:
Those are all very good points, and we should figure out what we should do, when we actually need them. For the initial commit of this framework with the basic sanity tests, we don't need to create/delete namespaces, so I removed all that code from this patch. Once we start creating tests that do need namespaces we can figure out how to generate unique name spaces that won't collide if run in parallel.

Reply:
Once we run tests in parallel (which is desirable for e2e testing) we will need >2 random namespaces. @awels If this pr is only addressing serial running of e2e will you please update the title and description? Thanks.